polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -756,14 +756,14 @@ trait ValidationBackend {
 			}
 
 			match validation_result {
-				Err(ValidationError::InvalidCandidate(
-					WasmInvalidCandidate::AmbiguousWorkerDeath |
-					WasmInvalidCandidate::AmbiguousJobDeath(_),
+				Err(ValidationError::PossiblyInvalid(
+					PossiblyInvalidError::AmbiguousWorkerDeath |
+					PossiblyInvalidError::AmbiguousJobDeath(_),
 				)) if num_death_retries_left > 0 => num_death_retries_left -= 1,
-				Err(ValidationError::InvalidCandidate(WasmInvalidCandidate::JobError(_)))
+				Err(ValidationError::Invalid(WasmInvalidCandidate::JobError(_)))
```

Comment:
Since we retry in this case, it belongs under `PossiblyInvalid`.

Reply:
Oops, nice catch!